tests/trafficmanagement/circuitbreaker_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package trafficmanagement
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/istioio"
+)
+
+func TestCircuitBreaker(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(istioio.NewBuilder("tasks__traffic_management__circuit_breaking").
+			Add(
+				istioio.Script{Input: istioio.Path("scripts/circuitbreaker_test_setup.txt")},
+				istioio.MultiPodWait("istio-io-circuitbreaker"),
+				istioio.Script{
+					Input: istioio.Evaluate(istioio.Path("scripts/trip_circuitbreaker.txt"), map[string]interface{}{
+						"isSnippet":                false,
+						"inputTerminalFlag":        "",
+						"beforeCircuitBreakVerify": " 2>&1 | verify_circuit_breaking 80 100 0 20",
+						"afterCircuitBreakVerify":  " 2>&1 | verify_circuit_breaking 25 75 25 75",
+						"outputRedirectionCmd":     "2>&1",
+					}),
+					SnippetInput: istioio.Evaluate(istioio.Path("scripts/trip_circuitbreaker.txt"), map[string]interface{}{
+						"isSnippet":                true,
+						"inputTerminalFlag":        "-it",
+						"beforeCircuitBreakVerify": "",
+						"afterCircuitBreakVerify":  "",
+						"outputRedirectionCmd":     "",
+					}),
+				}).
+			Defer(istioio.Script{Input: istioio.Path("scripts/circuitbreaker_test_cleanup.txt")}).
```

Comment:
I'd add verification of the istio-proxy stats to make sure they are in a similar format to what the user is looking for.

Reply:
I see something similar in the actual tutorial.
```
$ kubectl exec $FORTIO_POD -c istio-proxy -- pilot-agent request GET stats | grep httpbin | grep pending
cluster.outbound|8000||httpbin.default.svc.cluster.local.circuit_breakers.default.rq_pending_open: 0
cluster.outbound|8000||httpbin.default.svc.cluster.local.circuit_breakers.high.rq_pending_open: 0
cluster.outbound|8000||httpbin.default.svc.cluster.local.upstream_rq_pending_active: 0
cluster.outbound|8000||httpbin.default.svc.cluster.local.upstream_rq_pending_failure_eject: 0
cluster.outbound|8000||httpbin.default.svc.cluster.local.upstream_rq_pending_overflow: 21
cluster.outbound|8000||httpbin.default.svc.cluster.local.upstream_rq_pending_total: 29
```
Is this what you meant by "istio-proxy stats" verification? For instance, verifying that count of `upstream_rq_pending_overflow` falls within some range besides parsing fortio output?

Reply:
Exactly :) The user is asked to run the command to look at the stats, so we just need to verify that the output looks similar.

Reply:
I just added verification for output format. The counts look very erratic. When I was running tests, the percentage of requests getting 503 before tripping ranged from 70% to 100%. After tripping, it was even more erratic. Consequently `upstream_rq_pending_overflow` range was wide. Setting the range to be too wide defeats the purpose of it where as setting it to a narrow range would make the test flaky.